apptest/sharding_test.go

```diff
@@ -0,0 +1,82 @@
+package apptest
```

Comment:
I'd suggest move all test cases to sub-directory, so they don't mix-up with test framework code.

Reply:
Done.